ardalis/pluralsight-ddd-fundamentals
# Update to .NET8.0 and update Nuget packages

This PR updates the target framework to .NET8.0. Also, it updates all Nuget packages and fixes breaking changes.

* Remove deprecated MediatR.Extensions.Microsoft.DependencyInjection
* Update Ardalis.Endpoint and fix breaking changes
    - BaseAsyncEndpoint was renamed to EndpointBaseAsync
* Update SharedKernel and fix breaking changes
    - IApplicationEvent was renamed to BaseIntegrationEvent
* Update Ardalis.HttpClientTestExtensions
    - GetAndDeserialize was renamed to GetAndDeserializedAsync
* Change MediatR registration with Autofac.
    - See: https://stackoverflow.com/a/75677259
* Fix broken unit tests

Relates to: #68 
Addresses: #66 #67
